lib/alchemy/upgrader/four_point_zero.rb

```diff
@@ -1,3 +1,68 @@
+require 'thor'
+
+class Alchemy::Upgrader::FourPointZeroTasks < Thor
+  include Thor::Actions
+
+  no_tasks do
+
+    def convert_available_contents_to_nestable_elements
+      old_config_file = Rails.root.join('config', 'alchemy', 'elements.yml')
+      print "Reading config/alchemy/elements.yml ... "
+      elements_config = YAML.load_file(old_config_file)
+      puts "done."
+
+      print "Copying existing config file to config/alchemy/elements.yml.old ... "
+      FileUtils.copy old_config_file, Rails.root.join('config', 'alchemy', 'elements.yml.old')
+      puts "done."
+
+      print "Identifying `available_contents` ... "
+      elements_with_available_contents = elements_config.select { |e| e['available_contents'] }
+      puts 'done.'
+
+      print 'Converting to `nestable_elements` ... '
+      new_elements = elements_with_available_contents.inject([]) do |new_elements, old_element|
+        new_elements << modify_old_element(old_element.dup)
+        build_new_elements(old_element).inject(new_elements) do |ne, element_from_contents|
+          ne << element_from_contents
+        end
+      end.uniq.sort {|a, b| a['name'] <=> b['name'] }
```

Comment:
Style/MultilineBlockChain: Avoid multi-line chains of blocks.<br>Style/SpaceInsideBlockBraces: Space between { and | missing.